filters/apiusagemonitoring/spec.go

```diff
@@ -154,7 +158,31 @@ func (s *apiUsageMonitoringSpec) Name() string {
 	return filters.ApiUsageMonitoringName
 }
 
-func (s *apiUsageMonitoringSpec) CreateFilter(args []interface{}) (filter filters.Filter, err error) {
+func keyFromArgs(args []interface{}) string {
+	var sb strings.Builder
+	for _, a := range args {
+		s, ok := a.(string)
+		if !ok {
+			sb.Reset()
+			break
+		}
+		sb.WriteString(s)
+	}
+	return sb.String()
+}
+
+func (s *apiUsageMonitoringSpec) CreateFilter(args []interface{}) (filters.Filter, error) {
+	key := keyFromArgs(args)
+	if key != "" {
+		s.mu.Lock()
+		f, ok := s.filterMap[key]
+		if ok {
+			s.mu.Unlock()
+			return f, nil
```

Comment:
I don't like that filter instance references pathInfo which is mutable but I guess its ok in this case.